include/SFML/System/Time.inl

```diff
@@ -30,52 +30,68 @@ m_microseconds(0)
 }
 
 
+////////////////////////////////////////////////////////////
+template <typename Rep, typename Period>
+constexpr Time::Time(const std::chrono::duration<Rep, Period>& duration) :
+m_microseconds(duration)
+{
+}
+
+
 ////////////////////////////////////////////////////////////
 constexpr float Time::asSeconds() const
 {
-    return static_cast<float>(static_cast<double>(m_microseconds) / 1000000.0);
+    return std::chrono::duration<float>(m_microseconds).count();
 }
 
 
 ////////////////////////////////////////////////////////////
 constexpr Int32 Time::asMilliseconds() const
 {
-    return static_cast<Int32>(m_microseconds / 1000);
+    return std::chrono::duration_cast<std::chrono::duration<Int32, std::milli>>(m_microseconds).count();
 }
 
 
 ////////////////////////////////////////////////////////////
 constexpr Int64 Time::asMicroseconds() const
+{
+    return m_microseconds.count();
+}
+
+
+////////////////////////////////////////////////////////////
+constexpr std::chrono::microseconds Time::toDuration() const
 {
     return m_microseconds;
 }
 
 
 ////////////////////////////////////////////////////////////
-constexpr Time::Time(Int64 microseconds) :
-m_microseconds(microseconds)
+template <typename Rep, typename Period>
+constexpr Time::operator std::chrono::duration<Rep, Period>() const
 {
+    return m_microseconds;
 }
 
 
 ////////////////////////////////////////////////////////////
 constexpr Time seconds(float amount)
 {
-    return Time(static_cast<Int64>(amount * 1000000));
+    return Time(std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::duration<float>(amount)));
```

Comment:
Is this the most concise that constructing from seconds gets? I looked for some [other answers](https://stackoverflow.com/questions/17899684/convert-seconds-as-double-to-stdchronoduration), but they assume `duration<double>`.

The Chrono API is really awful... Who thought it was a good idea to keep `duration` generic? 😕

Reply:
Converting from floating point seconds to integer microseconds is lossy so the <chrono> API makes that look ugly. The syntax is much cleaner when lossy conversions aren't required. I suppose we could still do something like
```
return Time(std::chrono::microseconds(amount * 1'000'000));
```
but my goal was to remove all those magic numbers.

Reply:
I see, thanks for the explanation. It's weird to me that one of the universally most common operations of any Time API -- converting between seconds and durations -- is made so extremely hard. But it's maybe a case of APIs designed by people who don't need them for everyday productive use 😬

Anyway I'm glad we still keep the `sf::Time` class, it's very nice to work with.

Reply:
`sf::Time` is ergonomic because it's less correct. If we removed lossy conversions from `sf::Time` and made accept a generic type as the underlying representation, we'd quickly approach an interface as verbose as `<chrono>`. If you really want shorter code, we can liberally apply `using name std::chrono` within select scopes to minimize the character count.

Reply:
Forgot to answer here (and it's not direct feedback to this PR anyway, more interested in the discussion 🙂)

> If we removed lossy conversions from `sf::Time` and made accept a generic type as the underlying representation, we'd quickly approach an interface as verbose as `<chrono>`.

This is exactly my point: making `chrono::duration` generic was a design mistake in my opinion. No other language has generic durations and especially "business software" languages like Java need to deal _a lot_ with time (it also took them several attempts to get things finally right, but the current `Instant`/`Duration` API is pretty decent).

I don't know what the premise was, maybe someone thought the API must be capable of handling simulations of the Large Hadron Collider. I find it sometimes a bit sad that so many C++ APIs are designed to fit the 1% use cases while making the 99% extra hard 😕